spirv_msl.cpp

```diff
@@ -771,12 +782,58 @@ string CompilerMSL::to_function_args(uint32_t img, const SPIRType &imgtype, bool
 			remove_duplicate_swizzle(coord_y);
 			string coord_z = coord_expr + ".z";
 			remove_duplicate_swizzle(coord_z);
-			tex_coords = "float3(" + coord_x + ", (1.0 - " + coord_y + "), " + coord_z + ")";
+			if (is_fetch)
+				tex_coords = "uint3(" + coord_x + ", (1.0 - " + coord_y + "), " + coord_z + ")";
+			else
+				tex_coords = "float3(" + coord_x + ", (1.0 - " + coord_y + "), " + coord_z + ")";
 		}
 		else
 		{
 			tex_coords = coord_expr + ".xyz";
 			remove_duplicate_swizzle(tex_coords);
+
+			if (is_fetch)
+				tex_coords = "uint3(" + tex_coords + ")";
+		}
+
+		alt_coord = ".w";
+
+		break;
+
+	case DimCube:
+		if (msl_config.flip_frag_y)
```

Comment:
You're right. It is normally set to false. It was added originally because of the difference between texture coordinate systems. So far it hasn't been needed...and at some point it should come out...but I'm not convinced we have hit all the test cases yet with MSL.